src/fsharp/IlxGen.fs

```diff
@@ -3786,36 +3777,27 @@ and GenLambdaClosure cenv (cgbuf:CodeGenBuffer) eenv isLocalTypeFunc selfv expr
                 let ilContractTy = mkILFormalBoxedTy ilContractTypeRef ilContractGenericParams
                 let ilContractCtor =  mkILNonGenericEmptyCtor None cenv.g.ilg.typ_Object
 
-                let ilContractMeths = [ilContractCtor; mkILGenericVirtualMethod("DirectInvoke",ILMemberAccess.Assembly,ilContractMethTyargs,[],mkILReturn ilContractFormalRetTy, MethodBody.Abstract) ]
-
+                let ilContractMeths = [ilContractCtor; mkILGenericVirtualMethod("DirectInvoke",MethodAttributes.Assembly,ilContractMethTyargs,[],mkILReturn ilContractFormalRetTy, MethodBody.Abstract) ]
+                let ilContractTypeRefAttribute = ComputeTypeAccess ilContractTypeRef true
                 let ilContractTypeDef = 
                     { Name = ilContractTypeRef.Name 
                       Layout = ILTypeDefLayout.Auto
-                      Access =  ComputeTypeAccess ilContractTypeRef true
+                      Attributes =  (if ilContractTypeRefAttribute &&& TypeAttributes.HasSecurity <> enum 0 then ilContractTypeRefAttribute ^^^ TypeAttributes.HasSecurity else ilContractTypeRefAttribute) ||| TypeAttributes.Abstract ||| TypeAttributes.Serializable ||| TypeAttributes.SpecialName ||| TypeAttributes.BeforeFieldInit ||| TypeAttributes.AutoClass // the contract type is an abstract type and not sealed
```

Comment:
This line is long - put it in a separate let and use appropriate ilTypeDef.WithFooBar(..) update methods instead?